crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -157,15 +160,34 @@ fn make_executor(kind: ExecutorKind) -> Box<dyn SystemExecutor> {
 }
 
 /// Chain systems into dependencies
-#[derive(PartialEq)]
-pub enum Chain {
-    /// Run nodes in order. If there are deferred parameters in preceeding systems a
-    /// [`apply_deferred`] will be added on the edge.
-    Yes,
-    /// Run nodes in order. This will not add [`apply_deferred`] between nodes.
-    YesIgnoreDeferred,
-    /// Nodes are allowed to run in any order.
-    No,
+#[derive(Default)]
+pub struct Chain(
```

Comment:
Ah okay, I'm starting to see what's going on here.

I think that rather than storing an option, this would be clearer as something like:

```rust
pub enum Chain {
   Unchained,
   Chained(ChainConfigMap)
}

pub struct ChainConfigMap(BTreeMap<TypeId, Box<dyn Any>);
```
This then lets us get rid of the `is_chained` method: users can just use `matches!`.